### ISSUE TITLE
[Bug]: `useState`/`useArgs` in story is very laggy

### ISSUE THREAD

**Issue by emlai** [storybookjs/storybook#24442]
### Describe the bug

A controlled component (using `useState` or `useArgs`) is laggy/sluggish/slow to update in storybook:

https://github.com/storybookjs/storybook/assets/7543552/5281a268-4e35-4861-81ab-bac7d4ba0035

When opening the same story canvas in a new tab (aka iframe.html), it's smooth/fast as expected:

https://github.com/storybookjs/storybook/assets/7543552/c346a018-84d8-4a9c-a436-eccd744066b8

### To Reproduce

Make a story that uses `useState`/`useArgs` for state management, and use a component that calls `onChange` frequently, such as a Slider component, or other component that calls onChange each time the mouse is moved. For example:

```tsx
export const Default = {
  render: args => {
    const [value, setValue] = useState(0);
    return (
      <Slider {...args} value={value} onChange={value => setValue(value)} />
    );
  },
}
```

(let me know if a stackblitz repro is needed, and I'll make one) 

### System

```bash
System:
    OS: macOS 13.5
    CPU: (12) arm64 Apple M2 Pro
  Binaries:
    Node: 20.5.1 - /opt/homebrew/bin/node
    npm: 9.8.0 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 118.0.5993.70
    Safari: 16.6
```

**Comment by Dron007** [comment]
I confirm it in Windows 10 in Chrome. I am documenting integer field component with standard numeric input and the following render:
```
  render: function Render (args) {
    const [, setArgs] = useArgs()

    const onChange = (val) => {
      args.onChange(val)
      setArgs({ value: val })
    }

    return (
      <IntegerField
        {...args}
        onChange={onChange}
      />
    )
  },
```
and it is very slow on auto-repeat with arrow Up/Down keys. It freezes Chrome and slowly adds actions to the tab. When I open it in separate tab it works fast. Any component update after changing its source code is quite slow too.

**Comment by vanessayuenn** [comment]
@emlai yes a stackblitz repro would be very helpful! can you also confirm this is an issue across browsers? thank you!

**Comment by morriq** [comment]
I can confirm it. 

It's easy to reproduce with cpu throtling on every story using useArgs
https://developer.chrome.com/docs/devtools/settings/throttling

**Comment by anthonyma94** [comment]
@ndelangen any updates to this? I'm suffering from this as well. It's frustrating that this issue has been open for 2 years with no updates. I have to disable args at this point because it's unusable.
